registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/validator/impl/BiometricsSignatureValidator.java

```diff
@@ -0,0 +1,165 @@
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.kernel.biometrics.commons.BiometricsSignatureHelper;
+import io.mosip.kernel.biometrics.entities.BIR;
+import io.mosip.kernel.biometrics.entities.BiometricRecord;
+import io.mosip.kernel.biometrics.entities.Entry;
+import io.mosip.kernel.core.exception.BiometricSignatureValidationException;
+import io.mosip.kernel.core.http.RequestWrapper;
+import io.mosip.kernel.core.http.ResponseWrapper;
+import io.mosip.kernel.core.util.DateUtils;
+import io.mosip.kernel.core.util.StringUtils;
+import io.mosip.registration.processor.core.code.ApiName;
+import io.mosip.registration.processor.core.constant.JsonConstant;
+import io.mosip.registration.processor.core.constant.ProviderStageName;
+import io.mosip.registration.processor.core.exception.ApisResourceAccessException;
+import io.mosip.registration.processor.core.exception.PacketManagerException;
+import io.mosip.registration.processor.core.packet.dto.FieldValue;
+import io.mosip.registration.processor.core.packet.dto.JWTSignatureVerifyRequestDto;
+import io.mosip.registration.processor.core.packet.dto.JWTSignatureVerifyResponseDto;
+import io.mosip.registration.processor.core.spi.restclient.RegistrationProcessorRestClientService;
+import io.mosip.registration.processor.core.status.util.StatusUtil;
+import io.mosip.registration.processor.packet.storage.utils.PriorityBasedPacketManagerService;
+
+@Component
+public class BiometricsSignatureValidator {
+
+	private static final String DATETIME_PATTERN = "mosip.registration.processor.datetime.pattern";
+	
+	/** The Constant TRUE. */
+	private static final String TRUE = "true";
+	
+	/** The Constant EXCEPTION. */
+	private static final String EXCEPTION = "EXCEPTION";
+
+	@Autowired
+	private PriorityBasedPacketManagerService packetManagerService;
+
+	@Autowired
+	private RegistrationProcessorRestClientService<Object> registrationProcessorRestService;
+
+	@Autowired
+	private Environment env;
+
+	@Autowired
+	ObjectMapper mapper;
+
+	@Value("#{T(java.util.Arrays).asList('${mosip.regproc.common.before-cbeff-others-attibute.reg-client-versions:}')}")
+	private List<String> regClientVersionsBeforeCbeffOthersAttritube;
+
+	public void validateSignature(String id, String process, BiometricRecord biometricRecord)
+			throws JSONException, JsonParseException, JsonMappingException, JsonProcessingException, IOException,
+			BiometricSignatureValidationException, ApisResourceAccessException, PacketManagerException,
+			io.mosip.kernel.core.util.exception.JsonProcessingException {
+
+		// backward compatibility check
+		String version = getRegClientVersionFromMetaInfo(id, process);
+		if (!regClientVersionsBeforeCbeffOthersAttritube.contains(version)) {
```

Comment:
Please invert the condition and return to avoid nesting below code

Reply:
inverted this method.